trojsten/rules/susi.py

```diff
@@ -98,24 +106,24 @@ def prepare_coefficients(self, round):
             .values_list("user", "camps")
         )
 
+        self.number_of_solved_tasks = dict(
+            Submit.objects.filter(
+                Q(task__round__semester__competition=round.semester.competition, points__gt=0),
+                Q(task__round__semester__year__lt=round.semester.year)
+                | Q(
+                    task__round__semester__year=round.semester.year,
+                    task__round__semester__number__lt=round.semester.number,
+                ),
+            )
+            .values("user")
+            .annotate(solved_tasks=Count("task", distinct=True))
+            .values_list("user", "solved_tasks")
```

Comment:
Toto počíta všetky vyriešené úlohy dohromady, my by sme ale chceli počítať počet semestrov s 8 vyriešenými úlohami. Teda ak v dvoch častiach vyriešim po 5 úloh, koeficient sa nezmení, ale ak vyriešim v dvoch po 10 úloh, tak sa zvýši dvakrát. Ak vyriešim v jednej časti 20, tak sa zvýši iba raz.

Reply:
Aha tak takto som tú zmenu nepochopil, ale zmením to teda aby to rátalo časti s >=8 vyriešenými úlohami.

Reply:
Tu mám ale potom podobný problém ako pri kms, že to neviem inak spraviť ako iterovať cez semestre. Existuje nejaká group_by metóda pre django query sety?

Reply:
Nejaké group_by Django pozná. V KMS je problém v tom, že súčet bodov nie je tak ľahko vytiahnuteľný z DB, takže tam to asi o moc lepšie nejde. Tu sa to možno dá počítať zo submitov.

Reply:
Myslim ze som na to prisiel, skus sa na to pozriet.

Reply:
Rovnako vsak ako pri KMS, mozno sa tam niekde da pouzit rozumne prefetch_related alebo select_related aby to zbehlo rychlejsie.

Reply:
skusal som sa s tym pohrat ale tu uz je to spravene najrozumnejsie ako ide co sa tyka database access.